### ISSUE TITLE
Non-linear evaluation

### ISSUE THREAD

**Issue by f-hafner** [odissei-lifecourse/life-sequencing-dutch#113]
Replace the current linear regressions with some of these options
1. fine-tune existing model (fine-tune the LLM weights)
2. random forest or similar on the embeddings 
3. neural network on top of the LLM

**Comment by f-hafner** [comment]
duplicate of #97?

**Comment by tanzir5** [comment]
From #97 

I can unfreeze some latest layers of the LLM, add a linear layer in front of the model, and train the entire thing for specific tasks.

This is important because a linear regression model cannot capture interaction between the embedding features and possible non-linear transformations. With this setup, the existing model can actually help in doing both.

There is the issue of catastrophic forgetting where the model completely forgets about previous training and overtrains on the current train set for the current task. This can be circumvented by having a low enough learning rate, as well as by unfreezing as few layers as needed.

**Comment by tanzir5** [comment]
Simple NN on top of LLM has been implemented.

See also #144

**Comment by f-hafner** [comment]
closing as mostly completed
